index.bs

```diff
@@ -5401,6 +5405,60 @@ authenticator=] and [=client device=] SHOULD work together to ensure that the ex
 only to the operating system user that created that [=platform credential=].
 
 
+## Username Enumeration ## {#sctn-username-enumeration}
+
+While initiating a [=registration ceremony|registration=] or [=authentication ceremony=], there is a risk that the [=[WRP]=] might leak sensitive
+information about its registered users. For example, if a [=[RP]=] uses e-mail addresses as usernames and a user attempts to
+initiate an [=authentication=] [=ceremony=] for "alex.p.mueller@example.com" and the [=[RP]=] responds with a failure, but then
+successfully initiates an [=authentication ceremony=] for "j.doe@example.com", then the user can conclude that "j.doe@example.com"
+is registered and "alex.p.mueller@example.com" is not. The [=[RP]=] has thus leaked the possibly sensitive information that
+"j.doe@example.com" has an account at this [=[RP]=].
+
+The following is a non-normative, non-exhaustive list of measures the [=[RP]=] may implement to mitigate or prevent information
+leakage due to such an attack:
+
+- For [=registration ceremonies=]:
+
+    - If the [=[RP]=] uses [=[RP]=]-specific usernames as user identifiers:
+
+        - When initiating a [=registration ceremony=], disallow registration of usernames that are syntactically valid e-mail
+            addresses.
+
+            Note: The motivation for this suggestion is that in this case the [=[RP]=] probably has no choice but to fail the
+            [=registration ceremony=] if the user attempts to register a username that is already registered, and an information
+            leak might therefore be unavoidable. By disallowing e-mail addresses as usernames, the impact of the leakage can be
+            mitigated since it will be less likely that a user has the same username at this [=[RP]=] as at other [=[RPS]=].
+
+    - If the [=[RP]=] uses e-mail addresses as user identifiers:
+
+        - When initiating a [=registration ceremony=], interrupt the user interaction after receiving the e-mail address to
+            register and send a message to this address, containing an unpredictable one-time code and instructions for how to use
+            it to proceed with the ceremony. Display the same message to the user in the web interface regardless of the contents
+            of the sent e-mail and whether or not this e-mail address was already registered.
+
+            Note: This suggestion can be similarly adapted for other externally meaningful identifiers &mdash; for example,
+            national ID numbers or credit card numbers &mdash; if they provide similar out-of-band contact information &mdash; for
+            example, conventional postal address.
+
+- For [=authentication ceremonies=]:
+
+    - If, when initiating an [=authentication ceremony=], there is no account matching the provided username, continue the
+        ceremony by invoking `{{CredentialsContainer/get()|navigator.credentials.get()}}` using a syntactically valid
```

Comment:
hm, upon rendering, it looks like using backquotes does not work the way I was expecting. Perhaps we need to use `<code>...</code>` tags instead, or maybe not at all since the "bare" `{{AuthenticatorAssertionResponse}}` down below is rendering for me in monospace font.    (which has me wondering whether all the `<code>...</code>` tags we have in index.bs are needed any longer?)

Reply:
Ah, good catch. Fixed.

In expressions like `<code>|options|.{{CredentialCreationOptions/publicKey}}</code>` the `<code>` tags are necessary to get `|options|` and the periods to render with monospace font. It also does change the font size of `publicKey` slightly.

Reply:
Ah, ok, thx.